### ISSUE TITLE
[Feature Request] Export certain page range of document as pdf, png, etc

### ISSUE THREAD

**Issue by ParaN3xus** [Myriad-Dreamin/tinymist#826]
**Motivation**

For example, we have a long document(homework of the whole lesson), but we only want to export a specific part(homework of a chapter) of its content.

**Description**
User can specify page range to export in exporting tasks.

**Examples/Questions**
```jsonc
{
    // See https://go.microsoft.com/fwlink/?LinkId=733558
    // for the documentation about the tasks.json format
    "version": "2.0.0",
    "tasks": [
        {
            "label": "Export specific pages as PDF",
            "type": "typst",
            "command": "export",
            "export": {
                "format": "pdf",
                "page.range": "1..3", // or a list of pages to export
            }
        },
    ]
}
```
The argument and it's format is still open for disscussion.

**Comment by Eric-Song-Nop** [comment]
I personally think this form of defining which parts to export the document might not be so handy as you think. To know which pages to export, you still need the full document being processed, and you need to change the page numbers continuously as the document grows. For me, using some Typst functions to define which parts of your document should be used might be more flexible.

**Comment by ParaN3xus** [comment]
> To know which pages to export, you still need the full document being processed.

Yes, we are aware of this, because we will preview the entire document before exporting it. Actually this feature is simply to avoid the need to crop the desired pages again in other software after exporting the PDF.

> You need to change the page numbers continuously as the document grows.

We can use flexible range formats like "5.." or "5..-2" to export pages as the document grows, especially if the growth is at the end of the document. For more complex cases, we could use headings as boundaries, manually inserted metadata, or as you mentioned, define a function.
